test/tests/functional/pbs_qdel.py

```diff
@@ -164,3 +167,292 @@ def test_qdel_hstry_jobs_rerun(self):
         a = {'job_state': 'F', 'substate': '92'}
         self.server.expect(JOB, a, extend='x',
                            offset=1, id=jid, interval=1)
+
+    def test_qdel_history_job_rerun_nx(self):
+        """
+        Test rerunning a history job that was prematurely terminated due
+        to a a downed mom.
+        """
+        a = {'job_history_enable': 'True', 'job_history_duration': '5',
+             'job_requeue_timeout': '5', 'node_fail_requeue': '5',
+             'scheduler_iteration': '5'}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+        j = Job()
+        j.set_sleep_time(30)
+        jid = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid)
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'False'})
+        self.mom.stop()
+
+        # Force job to be prematurely terminated and try to delete it more than
+        # once.
+        try:
+            self.server.deljob([jid, jid, jid, jid])
+        except PbsDeljobError as e:
+            err_msg = "could not connect to MOM"
+            self.assertTrue(err_msg in e.msg[0],
+                            "Did not get the expected message")
+            self.assertTrue(e.rc != 0, "Exit code shows success")
+        else:
+            raise self.failureException("qdel job did not return error")
```

Comment:
unittest has an assert for this exact purpose.
Try:
```
with self.assertRaises(PbsDeljobError, msg="qdel job did not return error") as e:
    self.server.deljob([jid, jid, jid, jid])

self.assertIn(err_msg, e.exception.msg[0])
self.assertNotEqual(e.exception.rc, 0)
```

Reply:
Done